aslam_cv/aslam_cameras/include/aslam/cameras/implementation/PinholeProjection.hpp

```diff
@@ -777,9 +778,16 @@ bool PinholeProjection<DISTORTION_T>::initializeIntrinsics(const std::vector<Gri
     }
   }
 
-  //get the median of the guesses
-  if(f_guesses.empty())
-    return false;
+  if(f_guesses.empty()) {
+    double input_guess;
+    std::cout << "Initialization of focal length failed. Provide manual initialization: ";
+    std::cin >> input_guess;
```

Comment:
I would be very careful with asking the user suddenly from within some algorithm. I would instead add a default value as further argument to the algorithm. If that is to much work, what about reading a environment variable as such additional input?  E.g. `KALIBR_FOCAL_LENGTH_INIT_FALLBACK`. 

If it isn't set you can output a warning and return `false` (as before). This give the user the option to add such a value without breaking existing scripts in an particularly nasty way: make them run forever waiting for input.

Reply:
Thanks for the comment @HannesSommer. I decided to introduce a env variable to enable manual focal length input as it might be better to adjust it for each cameras etc.